com.chartboost.mediation/Android~/chartboost-mediation-android-bridge/src/main/java/com/chartboost/mediation/unity/BannerAdWrapper.kt

```diff
@@ -0,0 +1,345 @@
+            ad.load(placementName, size);
+        }
+    }
+
+    fun load(placementName:String, sizeName: String, sizeWidth:Float, sizeHeight:Float, x:Float, y:Float){
+        runTaskOnUiThread {
+            var size = HeliumBannerAd.HeliumBannerSize.STANDARD // default
+            when (sizeName) {
+                "ADAPTIVE" -> size = HeliumBannerAd.HeliumBannerSize.bannerSize(
+                    sizeWidth.roundToInt(),
+                    sizeHeight.roundToInt()
+                )
+
+                "STANDARD" -> size = HeliumBannerAd.HeliumBannerSize.STANDARD
+                "MEDIUM" -> size = HeliumBannerAd.HeliumBannerSize.MEDIUM
+                "LEADERBOARD" -> size = HeliumBannerAd.HeliumBannerSize.LEADERBOARD
+            }
+            createBannerLayout(size, x,y);
+            ad.load(placementName, size);
+        }
+    }
+    
+    fun setKeywords(keywords: Keywords) {
+        for (kvp in keywords.get()){
+            ad.keywords[kvp.key] = kvp.value
+        }
+    }
+    
+    fun setHorizontalAlignment(horizontalAlignment:Int) {
+        runTaskOnUiThread {
+            ad.foregroundGravity = when (horizontalAlignment) {
+                0 -> Gravity.LEFT
+                1 -> Gravity.CENTER_HORIZONTAL
+                2 -> Gravity.RIGHT
+                else -> Gravity.CENTER_HORIZONTAL
+            }
+        }
+    }
+
+    fun getHorizontalAlignment():Int {
+        return when (ad.foregroundGravity) {
+            Gravity.LEFT -> 0
+            Gravity.CENTER_HORIZONTAL -> 1
+            Gravity.RIGHT -> 2
+            else -> 1
+        }
+    }
+
+    fun setVerticalAlignment(verticalAlignment:Int) {
+        runTaskOnUiThread {
+            ad.foregroundGravity = when (verticalAlignment) {
+                0 -> Gravity.TOP
+                1 -> Gravity.CENTER_VERTICAL
+                2 -> Gravity.BOTTOM
+                else -> Gravity.CENTER_VERTICAL
+            }
+        }
+    }
+
+    fun getVerticalAlignment():Int {
+        return when (ad.foregroundGravity) {
+            Gravity.TOP -> 0
+            Gravity.CENTER_VERTICAL -> 1
+            Gravity.BOTTOM -> 2
+            else -> 1
+        }
+    }
+
+    fun getAdSize(): String? {
+        val size = ad.getSize();
+        val creativeSize = ad.getCreativeSizeDips();
+        
+        val json = JSONObject();
+        json.put("name", size?.name);
+        json.put("aspectRatio", size?.aspectRatio);
+        json.put("width", creativeSize.width);
+        json.put("height", creativeSize.height);
+        json.put("type", size?.isAdaptive);
+
+        return json.toString();
+    }
+
+    fun setDraggability(canDrag:Boolean) {
+        runTaskOnUiThread{
+            bannerLayout?.canDrag = canDrag
+        }
+    }
+
+    fun setVisibility(isVisible: Boolean) {
+        runTaskOnUiThread {
+            if (bannerLayout != null) {
+                val visibility = if (isVisible) View.VISIBLE else View.INVISIBLE
```

Comment:
Is this not a thing in Kotlin?

```suggestion
                val visibility = isVisible ? View.VISIBLE : View.INVISIBLE
```

Reply:
Nah, that doesn't work. If statements are expressions in Kotlin https://stackoverflow.com/questions/16336500/kotlin-ternary-conditional-operator